tests/integration/test_config.py

```diff
@@ -671,3 +672,44 @@ async def test_invalid_starttask(one_conf, flow, scheduler, start):
     with pytest.raises(InputError, match='a///b'):
         async with start(schd):
             pass
+
+
+async def test_CYLC_SOURCE_DIRECTORY_correctly_set(tmp_path, install, run_dir):
+    """CYLC_SOURCE_DIRECTORY is set correctly:
+
+    * In source dir
+    * In installed dir (Not testing different permutations of installed
+      dir as these are covered by testing of `get_workflow_source_dir`)
+    * Created directly in the run dir.
+
+    """
+    def mock_cylc_view(target):
+        """Run config through read_and_proc (~= cylc view --process)
+        """
+        return read_and_proc(
+            target,
+            viewcfg={
+                'mark': False,
+                'single': False,
+                'label': False,
+                'jinja2': True,
+                'contin': True,
+                'inline': True,
+            },
+        )
```

Comment:
We run a few CLI things from integration tests:
* Import the option parser function.
* Turn it into an "Options" object.
* Pull out the content of `def main` into `[async] def _main` as needed (to drop the CLI / asyncio stuff).